Modules/Product/Http/Controllers/ProductController.php

```diff
@@ -2,665 +2,292 @@
+                'data' => ""
+            ]);
         }
-        if (is_countable($variations) && count($variations) == 0) {
+        $response = $this->productService->update($request);
 
-            ProductVariation::where('product_id', $product_id)->update(array('status' => 2));
-        }
+        return response()->json($response);
+    }
 
-        //pre packs
-        if ($request->input('sell_type') == 3) {
-            $prePacks = json_decode($request->input('pre_packs'), true);
-            if ($prePacks) {
-                Products::where('id', $product_id)->update(array('sell_type' => '3'));
-                foreach ($prePacks as $prePack) {
-                    $active = $prePack['active'];
-                    if (isset($prePack['id']) && !empty($prePack['id'])) {
-                        if (isset($prePack['status']) && $prePack['status'] == 'deleted') {
-                            ProductPrepack::where('id', $prePack['id'])->delete();
-                        } else {
-                            $packsPrice = $prePack['packs_price'] && !in_array($prePack['packs_price'], array('', 'undefined', 'null')) ? $prePack['packs_price'] : 0;
-                            ProductPrepack::where('id', $prePack['id'])
-                                ->update([
-                                    'style' => $prePack['style'],
-                                    'pack_name' => $prePack['pack_name'],
-                                    'size_ratio' => $prePack['size_ratio'],
-                                    'size_range' => $prePack['size_range_value'],
-                                    'packs_price' => $packsPrice,
-                                    'active' => $active
-
-                                ]);
-                        }
-                    } else {
-                        $packsPrice = $prePack['packs_price'] && !in_array($prePack['packs_price'], array('', 'undefined', 'null')) ? $prePack['packs_price'] : 0;
-                        $ProductPrepack = new ProductPrepack();
-                        $ProductPrepack->product_id = $product_id;
-                        $ProductPrepack->style = $prePack['style'];
-                        $ProductPrepack->pack_name = $prePack['pack_name'];
-                        $ProductPrepack->size_ratio = $prePack['size_ratio'];
-                        $ProductPrepack->size_range = $prePack['size_range_value'];
-                        $ProductPrepack->packs_price = $packsPrice;
-                        $ProductPrepack->active = $active;
-                        $ProductPrepack->save();
-                    }
-                }
-            }
+    /**
+     * Change status like published, Unpublished of products
+     *
+     * @param Request $request
+     * @return JsonResponse
+     */
+    public function status(Request $request): JsonResponse
+    {
+        $user = auth()->user();
+        if ($user->cannot('viewAny', Product::class)) {
```

Comment:
Why "viewAny" for status change? this should be update

Reply:
Done